### ISSUE TITLE
Could we limit depth of reporting in summarize plan?

### ISSUE THREAD

**Issue by prjemian** [bluesky/bluesky#1254]
This question from an instrument scientist could be proposition for an enhancement to `summarize_plan()`.

## Proposition
That said, I was wondering if `summarize_plan()` could have a parameter which would enable a *new* kwarg called `level` which indicates how many calls deep (such as depth of `yield from plan()`)  to be shown in the summary. 

* level=0 (default) summarize as is now to level of moving motors etc.
* Level=1  summarize only current document, present all called functions by their names only (and do not go into summarizing internals of the called functions)
* Level=2  summarize current level and one level below, anything on lower levels is presented by their name

The use case is to aid in debugging of user-written code.  Most errors will happen in the user-written levels which might be only a couple calls deep.

Basically, if I write some data collection loops or other complicated fun stuff, I want to see what the loops will do but skip through details of the instrument's standard collection plans (SAXS/WAXS/USAXS).

**Comment by prjemian** [comment]
@jilavsky : You asked about this.  For this to happen, `summarize_plan()` would have to keep track of the level in the plan stack at which a given Message object was emitted.

Establishing a measure of the plan stack depth might also be useful for limiting depth of exception traces.

**Comment by tacaswell** [comment]
When we pull on a generator we don't have an (easy) way to tell what the stack level the function you want to ID is (as the stack in the generator will be where ever the the actual `yield` is).  This is mostly a feature that the internal details of how the plan constructs its self is hidden from the consumers (if it were not, the pre-processors would be much harder to get right).

I think a version of `summarize_plan` that suppresses messages between start / stop messages is probably the easier way to get the behavior you want.

**Comment by ambarb** [comment]
If a less verbose version is made, I would use it in the way @prjemian describes.  For now, I just change the number of points in the plan to 1 or 3, and the number of "loops" to 2 just to make sure the the flow control behavior is programmed correctly.   

On top of this, it would be interesting to make `summarize_plan` a wrapper with N output options for the N versions of `summarize_plan` that could exist.

**Comment by tacaswell** [comment]
```python
from bluesky import plans as pb
import bluesky.plan_stubs as bps
from ophyd.sim import *

def my_nested_plan(dets, mtrA, mtrB):
    for i in range(15):
        yield from bps.mv(mtrA, i)
        yield from bp.count(dets, num=15)
        yield from bp.scan(dets, mtrB, -1, 1, 25)


      
```


```python
In [215]: summarize_plan(my_nested_plan([], motor, motor1))
motor -> 0
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 1
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 2
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 3
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 4
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 5
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 6
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 7
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 8
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 9
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 10
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 11
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 12
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 13
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
motor -> 14
  count(detectors=[], num=15)
  scan(detectors=[], num=25, args=["SynAxis(prefix='', name='motor1', read_attrs=['readback', 'setpoint'], configuration_attrs=['velocity', 'acceleration'])", -1, 1], per_step=None)
```

**Comment by stan-dot** [comment]
this is still a problem, solution needs a second look and definitely a set of unit tests
